cmd/esc/cli/env_clone.go

```diff
@@ -37,17 +37,23 @@ func newEnvCloneCmd(env *envCommand) *cobra.Command {
 				return err
 			}
 
-			ref := env.parseRef(args[0])
-			if ref.isUsingLegacyID {
-				return errors.New("invalid source environment ID")
+			ref, args, err := env.getExistingEnvRef(ctx, args)
+			// An error will arise if an environment reference is ambiguous and can't be
+			// resolved to a single environment. The ref for the non-legacy environment will
+			// also be returned in this case.
+			// If the original ref is using a legacy ID of just env name return an error
+			// Otherwise we will ignore any conflict errors and assume the user meant <project-name>/<env-name>
+			if (err != nil && ref.isUsingLegacyID) || ref.isUsingLegacyID {
```

Comment:
let's just simplify this to `if ref.isUsingLegacyID` (this is for both cases for err is nil or not), and make sure to add an `if err != nil` check after this conditional for the case where there is a non-legacy id related error